dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/udp/UDPConfigurationImpl.java

```diff
@@ -0,0 +1,467 @@
+package io.openliberty.netty.internal.udp;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.ibm.websphere.channelfw.ChannelData;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelOption;
+import io.openliberty.netty.internal.BootstrapConfiguration;
+import io.openliberty.netty.internal.exception.NettyException;
+
+/**
+ * TODO: as far as I can tell UDP options aren't actually exposed to user config in Liberty, so currently a few props 
+ * (notably access lists) aren't implemented.
+ * 
+ * @see com.ibm.ws.udpchannel.internal.UDPChannelConfiguration
+ */
+public class UDPConfigurationImpl implements BootstrapConfiguration {
+
+    private static final TraceComponent tc = Tr.register(UDPConfigurationImpl.class, UDPMessageConstants.NETTY_TRACE_NAME, UDPMessageConstants.UDP_BUNDLE);
+
+    private static int DEFAULT_READ_BUFFER_SIZE = 1024000;
+    private Map<String, Object> channelProperties = null;
+
+
+    private ChannelData channelData;
+    private String hostname = null;
+    private int port = 0;
+    private int sendBufferSize = 0;
+    private int receiveBufferSize = 0;
+    private int channelReceiveBufferSize = UDPConfigConstants.MAX_UDP_PACKET_SIZE;
+    private String addressExcludeList[] = null;
+    private String addressIncludeList[] = null;
+    private int retryCount = 10;
+    private int retryInterval = 5000;
+    private String externalName = null;
+
+    private boolean isInbound;
+
+    /**
+     * Constructor.
+     * @throws NettyException 
+     */
+    public UDPConfigurationImpl(Map<String, Object> options, boolean inbound) throws NettyException {
+        this.sendBufferSize = DEFAULT_READ_BUFFER_SIZE;
+        this.receiveBufferSize = DEFAULT_READ_BUFFER_SIZE;
+        this.channelProperties = options;
+        isInbound = inbound;
+        if (this.channelProperties != null) {
+            // read in values now, to save time reading them in each time
+            // they are requested
+            setValues(channelProperties);
+        } else {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(tc, "UDPConfigurationImpl object constructed with null properties");
+            }
+            throw new NettyException("UDPConfigurationImpl constructed with null properties");
+        }
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            outputConfigToTrace();
+        }
+    }
+
+    /**
+     * @see com.ibm.ws.udpchannel.internal.UDPNetworkLayer.initDatagramSocket(VirtualConnection)
+     */
+    @Override
+    public void applyConfiguration(Bootstrap bootstrap) {
+        // TODO: UDPNetworkLayer sets reuseaddr to false, but if we don't do that here we can't bind TCP and UDP to the same port...?
```

Comment:
Convert to issue?

Reply:
https://github.com/OpenLiberty/open-liberty/issues/18520